pkg/api/types.go

```diff
@@ -1014,6 +1014,9 @@ type LiteralTestStep struct {
 	// RunAsScript defines if this step should be executed as a script mounted
 	// in the test container instead of being executed directly via bash
 	RunAsScript *bool `json:"run_as_script,omitempty"`
+	// NodeArchitecture is the architecture for the node where the test will run.
+	// If set, the generated test pod will include a nodeSelector for this architecture.
+	NodeArchitecture NodeArchitecture `json:"node_architecture,omitempty"`
```

Comment:
It looks like most of the fields here are pointers, is there a reason why you chose not to use a pointer here?

Reply:
`NodeArchitecture` was defined on line 782 and I chose to use as it is.

Reply:
@smg247 do you want me to make it a pointer?

Reply:
yeah, I think it would be simpler if you did:
```suggestion
	NodeArchitecture *NodeArchitecture `json:"node_architecture,omitempty"`
```